src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,245 @@
 
 #include "SubWindow.h"
 
+#include <QMdiArea>
 #include <QMoveEvent>
 #include <QResizeEvent>
 #include <QWidget>
 
+#include "embed.h"
+#include <QDebug>
+
 
 SubWindow::SubWindow(QWidget *parent, Qt::WindowFlags windowFlags)
   : QMdiSubWindow(parent, windowFlags)
 {
 	// initialize the tracked geometry to whatever Qt thinks the normal geometry currently is.
 	// this should always work, since QMdiSubWindows will not start as maximized
-	m_trackedNormalGeom = normalGeometry();
+  	m_trackedNormalGeom = normalGeometry();
+
+    // inits the colors
+    m_activeColor = Qt::SolidPattern;
+    m_textShadowColor = Qt::black;
+  	m_borderColor = Qt::black;
+
+    //close, minimize, maximize and restore(after minimize) buttons
+    m_closeBtn = new QPushButton( embed::getIconPixmap( "close" ), QString::null, this );
+    m_closeBtn->resize(17, 17);
+    m_closeBtn->setFocusPolicy(Qt::NoFocus);
+    m_closeBtn->setToolTip(tr("Close"));
+    connect(m_closeBtn, SIGNAL(clicked(bool)),this,SLOT(close()) );
+
+    m_maximizeBtn = new QPushButton( embed::getIconPixmap( "maximize" ), QString::null, this );
+    m_maximizeBtn->resize(17, 17);
+    m_maximizeBtn->setFocusPolicy(Qt::NoFocus);
+    m_maximizeBtn->setToolTip(tr("Maximize"));
+    connect(m_maximizeBtn, SIGNAL(clicked(bool)),this,SLOT(showMaximized()));
+
+    m_minimizeBtn = new QPushButton( embed::getIconPixmap( "minimize" ), QString::null, this );
+    m_minimizeBtn->resize(17, 17);
+    m_minimizeBtn->setFocusPolicy(Qt::NoFocus);
+    m_minimizeBtn->setToolTip(tr("Minimize"));
+    connect(m_minimizeBtn, SIGNAL(clicked(bool)),this,SLOT(showMinimized()));
+
+    m_restoreBtn = new QPushButton( embed::getIconPixmap( "restore" ), QString::null, this );
+    m_restoreBtn->resize(17, 17);
+    m_restoreBtn->setFocusPolicy(Qt::NoFocus);
+    m_restoreBtn->setToolTip(tr("Restore"));
+    connect(m_restoreBtn, SIGNAL(clicked(bool)),this,SLOT(showNormal()));
```

Comment:
I know we keep one eye shut on the 'space between parentheses' rule, but you should really put a space after your commas.